src/components/SearchBar.jsx

```diff
@@ -1,64 +1,94 @@
-import { useEffect, useState } from "react"
-import { getArticles, getTopics } from "../api"
-import Loading from "./Loading"
+import { useEffect, useState } from 'react';
+import { getArticles, getTopics } from '../api';
+import Loading from './Loading';
 
-function SearchBar({setArticlesList}) {
- 
-  const [topicsList, setTopicsList] = useState()
-  const [topicQuery, setTopicQuery] = useState()
-  const [isLoading, setIsLoading] = useState(true)
-	const [error, setError] = useState(false)
+function SearchBar({ setArticlesList }) {
+	const [topicsList, setTopicsList] = useState();
+	const [topicQuery, setTopicQuery] = useState();
+	const [sortQuery, setSortQuery] = useState();
+	const [orderQuery, setOrderQuery] = useState();
+	const [isLoading, setIsLoading] = useState(true);
+	const [error, setError] = useState(false);
 
-  useEffect(() => {
+	useEffect(() => {
     setError(false)
-    getTopics().then((topics) => {
-      setTopicsList([{slug: '' }, ...topics])
-      setIsLoading(false)
-    }).catch((err) => {
-      setError(true);
-      setIsLoading(false);
-    });
-  }, [topicsList])
+		getTopics()
+			.then((topics) => {
+				setTopicsList([{ slug: '' }, ...topics]);
+				setIsLoading(false);
+			})
+			.catch((err) => {
+				setError(err);
+				setIsLoading(false);
+			});
+	}, [topicsList]);
```

Comment:
Having this state in the dependency array will trigger the useEffect to run again... which sets the state again... which triggers the useEffect again etc! Pop a console.log into this component - are we getting any looping?